kratos/mpi/sources/mpi_data_communicator.cpp

```diff
@@ -474,6 +474,32 @@ bool MPIDataCommunicator::IsNullOnThisRank() const
     return mComm == MPI_COMM_NULL;
 }
 
+const DataCommunicator& MPIDataCommunicator::GetSubDataCommunicator(
+    const std::vector<int>& rRanks,
+    const std::string& rNewCommunicatorName
+    ) const
+{
+    if (ParallelEnvironment::HasDataCommunicator(rNewCommunicatorName)) {
+        const DataCommunicator& r_data_communicator = ParallelEnvironment::GetDataCommunicator(rNewCommunicatorName);
+        const int rank = Rank();
+        const auto it_find = std::find(rRanks.begin(), rRanks.end(), rank);
+        std::size_t number_of_active_ranks = 0;
+        if (it_find != rRanks.end()) {
+            KRATOS_ERROR_IF_NOT(r_data_communicator.IsDefinedOnThisRank()) << "The rank " << rank << " does not participate in the existing data communicator " << rNewCommunicatorName  << " despite being in the provided rank list" << std::endl;
+        } else {
+            number_of_active_ranks = 1;
```

Comment:
Why are you setting it to 1 here? I would think that the rank should not participate in the new communicator if we reach the `else` branch

Reply:
I put the one oin the other side, silly me :P

Reply:
Now